packages/dart/noports_core/lib/src/npt/npt.dart

```diff
@@ -313,4 +321,29 @@ class _NptImpl extends NptBase
 
     return localRvPort;
   }
+
+  @override
+  Future<SocketConnector> runInline() async {
+    int localRvPort = await _preRun();
+    sendProgress('Creating connection to socket rendezvous');
+    if (!params.inline) {
+      logger.warning(
+          "WAT - runInline() was called but params.inline = false, running under the assumption that params.inline was meant to be true.");
+    }
```

Comment:
Putting this warning in place seemed more sensible than introducing a layer of abstraction to ensure type safety.